state-synchronizer/src/executor_proxy.rs

```diff
@@ -75,15 +74,18 @@ impl ExecutorProxyTrait for ExecutorProxy {
 
         let current_verifier = storage_info.get_validator_set().into();
 
-        let synced_trees = if let Some(synced_tree_state) = storage_info.synced_tree_state {
-            ExecutedTrees::from(synced_tree_state)
-        } else {
-            ExecutedTrees::from(storage_info.committed_tree_state)
-        };
+        let highest_synced_version =
+            if let Some(synced_tree_state) = storage_info.synced_tree_state {
+                ExecutedTrees::from(synced_tree_state)
+            } else {
+                ExecutedTrees::from(storage_info.committed_tree_state)
+            }
+            .version()
+            .unwrap_or(0);
```

Comment:
can you walk me through the case where this would not be set? cause even at the beginning of time it is... so this looks like it would mask a bug?

Reply:
would not be set?

Reply:
Why wouldn't it be set? Could you walk me through the cases when it would not be set? (i.e., none unwrap would return anything but a valid value)

Reply:
It seems not, let me change it to expect.

Reply:
I mean could we document the case when this wouldn't be set, what does that mean, should we actually assume that means we got a positive result or should we log an error?

Reply:
What's up?

Reply:
The version must exist since at least genesis should exist.